final_main/data/classes.js

```diff
@@ -18,4 +18,219 @@
 //     '41f83110b3a1eb1911a2eac'
 //   ],
 //   classCapacity: 20
-// }
\ No newline at end of file
+// }
+
+import { classes } from '../config/mongoCollections/js';
+import { ObjectId } from 'mongodb';
+import {
+    isValidAction,
+    isValidClassCapacity,
+    isValidClassName,
+    isValidDescription,
+    isValidId,
+    isValidTimeSlot,
+    isValidname,
+} from '../validateData.js';
+
+//get a class provided the id
+export const getClassbyId = async (id) => {
+    id = isValidId(id);
+    const classCollection = await classes();
+    const theClass = await classCollection.findOne({ _id: new ObjectId(id) });
+    if (theClass === null) { throw "No Class"; };
+    theClass._id = theClass._id.toString();
+    return theClass;
+}
+
+export const createClass = async (
+    className,
+    slots,
+    instructor,
+    description,
+    classCapacity
+) => {
+    className = isValidClassName(className);
+    slots = isValidTimeSlot(slots);
+    instructor = isValidname(instructor);
+    description = isValidDescription(description);
+    classCapacity = isValidClassCapacity(classCapacity);
+    let registeredUsers = [];
+    let reviews = [];
+
+    const classCollection = await classes();
+    const existingClass = await getAllClass();
+
+    // for (let i = 0; i < existingClass.length; i++) {
+
+    // }
+
+    newUser = {
+        className: className,
+        slots: slots,
+        instructor: instructor,
+        description: description,
+        classCapacity: classCapacity
+    }
+
+    const insertInfo = await classCollection.insertOne(newUser);
+    if (!insertInfo.acknowledged || !insertInfo.insertInfo) { throw 'Error: Could not create user'; };
+    const newId = insertInfo.insertId.toString();
+    const newClass = await getClassbyId(newId);
+    return newClass;
+}
+
+//get all the classes in the database
+export const getAllClass = async () => {
+    const classCollection = await classes();
+    let classList = await classCollection.find({}).toArray();
+    if (classList.length == 0) { return classList; };
+    if (!classList) { throw "Could not get all classes"; };
+    classList = classList.map((element) => {
+        element._id = element._id.toString();
+        return element;
+    });
+    return classList;
+}
+
+//delete a class
+export const deleteUser = async (id) => {
+    id = isValidId(id);
+    const classCollection = await classes();
+    const deletionInfo = await classCollection.findOneAndDelete({
+        _id: new ObjectId(id)
+    });
+
+    if (deletionInfo.lastErrorObject.n === 0) {
+        throw `Could not delete band with id of ${id}`;
+    }
+    return `${deletionInfo.value.firstName} ${deletionInfo.value.lastName} has been successfully deleted!`;
+};
+
+
+export const update = async (
```

Comment:
maybe rename this function also as updateclass() to have better readability while calling this function from other files